tree/ntuple/v7/test/ntuple_fork.hxx

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <functional>
+#include <string_view>
+
+// These helpers are split into an *Impl function and a macro to check for and propagate fatal failures to the caller.
+
+void ExecInForkImpl(std::function<void(void)> fn);
+
+void DeclarePointerImpl(std::string_view type, std::string_view variable, void *ptr);
+
+void ProcessLineImpl(const char *line);
+
+#ifdef NTUPLE_FORK_MACROS
```

Comment:
Why do we need to protect these macro definitions? But see also my other comment if we can just have all implementations in a header file